keps/sig-storage/625-csi-migration/README.md

```diff
@@ -156,10 +163,211 @@ Running this set of tests is optional for a per-provider basis. We would
+  - SLO with migration on matches the existing plugin's in-tree SLO with offset less than 1%
+
+* **Are there any missing metrics that would be useful to have to improve observability
+of this feature?**
+Node side CSI operation metrics. It will be implemented in the GA phase.
+
+### Dependencies
+
+
+* **Does this feature depend on any specific services running in the cluster?**
+
+  - Corresponding CSI Driver for in-tree CSI migration enabled plugin
+    - Usage description:
+      - Impact of its outage on the feature: in-tree plugin stops working without the CSI Driver properly setup
+      - Impact of its degraded performance or high-error rates on the feature: Error or performance decrease will be reflected for volume intensive operation
+
+
+### Scalability
+
+* **Will enabling / using this feature result in any new API calls?**
+  Yes.
+  - For volume attach/detach:
+    VolumeAttachment CREATE/PATCH APIs will be called for volume attachment by kube-controller-manager and csi-attacher. VolumeAttachment delete API will be called for volume detachment by csi-attacher. 1 for each needed.
+    VolumeAttachment LIST/WATCH api will be called by csi-attacher to monitor the VolumeAttachment.
+  - For volume provision/delete:
+    PVC LIST/WATCH apis will be called by csi-provisioner to monitor the PVC status. PV CREATE api will be called by csi-provisioner to create PV. PV DELETE api will be called by csi-provisioner to delete PV. PVC/PV PATCH api will be called by csi-provisioner for updating the object.
+    Notice that these new calls from csi-provisioner also mean that we will reduce call from the kube-controller-manager side.
+  - When CSI driver is being installed, there will be CSIDriver/CSINode CREATE/PATCH api call by kubelet. For each kubelet that installs the driver there will be one call.
+  - csi-provisioner && csi-attacher will call one time GET api for CSINode object when provision/attach volume
+
+* **Will enabling / using this feature result in introducing new API types?**
+  No
+
+* **Will enabling / using this feature result in any new calls to the cloud provider?**
+  After switching to CSI driver model, all the volume operations including volume provision/deletion/attach/detach/mount/unmount/resize will be running through the CSI driver. 
+  So depending on how the CSI driver is designed and implemented, it could vary if there is any new calls being added. 
+  For example, `gce-pd` driver has the in-tree and CSI version of plugin implementation for all the operation mentioned above, once we switch from in-tree to CSI by CSI migration. 
+  If the implementation is the same, then there will not be new calls to the cloud provider. 
+  However, it is also possible that the plugin maintainer has different implementation so there might be new calls.
+
+* **Will enabling / using this feature result in increasing size or count of the existing API objects?**
```

Comment:
Can you clarify which fields are specific for migration vs which fields are used for CSI in general?

Reply:
Done